openbsd/efivars.md

```diff
@@ -0,0 +1,150 @@
+# Instructions for getting user-space EFI vars API on OpenBSD
+
+Read [README.md](./README.md) first for general information.
+
+This is very similar to what's described in [esrt.md](./esrt.md) and
+[esrt-api.md](./esrt-api.md), but duplication is intentional for simplicity.
+
+The plan is:
+ * obtain `efivars-api` branch of https://github.com/3mdeb/openbsd-src fork of
+   OpenBSD source tree
+ * compile kernel
+ * compile EFI bootloader
+ * install both of them and create `/dev/efi`
+ * compile and install test tool that uses EFI vars API
+ * reboot and check that it works
+
+## Preparation
+
+As a regular user assuming `doas` is configured:
+
+```
+# install git package
+doas pkg_add git
+
+# add your user to wsrc and wobj groups (relogin to apply these changes)
+doas user mod -G wsrc,wobj $USER
+
+git clone --depth 1 -b efivars-api https://github.com/3mdeb/openbsd-src /usr/src
+```
+
+## Kernel compilation
+
+```
+# /sys/ is a symbolic link to /usr/src/sys/, go there to setup kernel build
+cd /sys/arch/amd64/conf/
+
+# generate configuration (the file is already there on the branch)
+config CUSTOM
+
+# build the kernel for the first time
+make -C ../compile/CUSTOM -j$(nproc)
```

Comment:
@SergiiDmytruk `nproc` was not found on my VM when followind your auto VM instructinos.
Is there `nproc` program on BSD at all?

I found to use `sysctl -n hw.ncpu` instead here, for example.

Reply:
I thought it's POSIX, but it's from GNU coreutils, so base OpenBSD system doesn't have it. Thanks.

Reply:
Confirmed updated document